Error processing GIBS responses in OPS

# Summary
After deployment of 0.2.3 to cumulus ops account, it was noticed that the messages deleted for `svc-bignbit-podaac-ops-cumulus-gibs-response-queue` remained at 0 but messages visible was climbing meaning GIBS is sending responses but they are not being processed.

The metrics for `svc-bignbit-podaac-ops-cumulus-handle_gitc_response` lambda shows 100% error rate

Looking at the logs for the lambda, the following stacktrace is found:
```
[ERROR] HTTPError: 404 Client Error: Not Found for url: https://cmr.uat.earthdata.nasa.gov/search/concepts/G3414659522-POCLOUD.umm_json
Traceback (most recent call last):
  File "/home/dockeruser/app/bignbit/handle_gitc_response.py", line 40, in handler
    umm_json = utils.get_umm_json(granule_concept_id, cmr_env)
  File "/home/dockeruser/app/bignbit/utils.py", line 139, in get_umm_json
    umm_json_response.raise_for_status()
  File "/home/dockeruser/app/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
```

OPS should not be querying `https://cmr.uat` 

Inspected the `CMR_ENVIRONMENT` variable for the `svc-bignbit-podaac-ops-cumulus-handle_gitc_response` lambda and observed it was incorrectly set to `UAT`.

A DLQ is configured for the GIBS response queue so all the messages that failed the lambda execution were moved to `svc-bignbit-podaac-ops-cumulus-gibs-response-dlq` as designed. There are currently about 12k messages on the DLQ.

# Actions Taken

- `CMR_ENVIRONMENT` was updated to `OPS` using the AWS dashboard for `svc-bignbit-podaac-ops-cumulus-handle_gitc_response` lambda
- Executed redrive to source queue for all messages on `svc-bignbit-podaac-ops-cumulus-gibs-response-dlq` 
- Observed error rate reduced to 0% after environment update ![Image](https://github.com/user-attachments/assets/a94ecad1-df87-4256-94b8-2f2e33650357)

# Root Cause Analysis

This check is not case insensitive:
https://github.com/podaac/bignbit/blob/develop/terraform/lambda_functions.tf#L446

And the podaac cumulus deployment is using lowercase `ops` for `local.environment`.

Suggested fix is to make the check case insensitive in the terraform code.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Deprecated
 ### Removed
 ### Fixed
+- [issues/60](https://github.com/podaac/bignbit/issues/60): Fixed bug causing GIBS responses to fail processing in OPS due to a case-sensitive comparison of environment name.
 ### Security
 
 ## [0.2.3]

--- a/terraform/lambda_functions.tf
+++ b/terraform/lambda_functions.tf
@@ -443,7 +443,7 @@ resource "aws_lambda_function" "handle_gitc_response" {
       REGION                      = data.aws_region.current.name
       BIGNBIT_AUDIT_BUCKET_NAME   = var.bignbit_audit_bucket
       BIGNBIT_AUDIT_PATH_NAME     = var.bignbit_audit_path
-      CMR_ENVIRONMENT             = local.environment != "OPS" ? "UAT" : ""
+      CMR_ENVIRONMENT             = upper(local.environment) != "OPS" ? "UAT" : ""
       EDL_USER_SSM                = var.edl_user_ssm
       EDL_PASS_SSM                = var.edl_pass_ssm
     }

